File: firstcode/phase gate one/gateOne/creditCardValidation/python/validationfunction.py
def double_every_second_number(credit_card_number):  
	total = 0
	for count in range(0, len(credit_card_number), 2):
		value = int(credit_card_number[count])
		value = (value * 2)

		if value >= 10:
			first_number = value % 10
			second_number = value // 10
			value = (first_number + second_number)
			total = total + value
		else:
			total = total + value
	
	return total



def sum_of_numbers_in_odd_position(credit_card_number):
	total = 0
	for count in range(0, len(credit_card_number)):
		value = int(credit_card_number[count]);
		if not count % 2 == 0:
			total = total + value
	return total
	
def card_type(credit_card_number):

	first_card = int(credit_card_number[0])
	second_card = int(credit_card_number[0])
	third_card_first_digit = int(credit_card_number[0])
	third_card_second_digit = int(credit_card_number[1])
	fourth_card_number = int(credit_card_number[0])
	
	if first_card == 4:
		return "VisaCard"

	elif second_card == 5:
		return "MasterCard"

	elif third_card_first_digit == 3 and third_card_second_digit == 7:
		return "American Express Cards"
	elif fourth_card_number == 6:
		return "Discover Cards"
	else:
		return "Invalid Card Type"

def card(credit_card_number):

	type_of_card = card_type(credit_card_number)
	first_value = double_every_second_number(credit_card_number)
	second_value = sum_of_numbers_in_odd_position(credit_card_number)
	
	length = len(credit_card_number)
	card_digit_length = str(length)
	
	sum_of_the_values = (first_value + second_value)
	
	if sum_of_the_values % 10 == 0:
		message =  "Credit Card Is Valid"
	else:
		message =  "Credit Card Is invalid"
	
	card_data = {"card_type": type_of_card, "card_number": credit_card_number, "card_digit_length": card_digit_length, "card_status": message}
	
	return card_data

File: firstcode/phase gate one/gateOne/creditCardValidation/java/ValidationFunction.java
public class ValidationFunction {
	
	public int doubleEverySecondNumber(String creditCardNumber) {                 
		final int STRING_LENGTH = creditCardNumber.length();  		      
		
		int count = (STRING_LENGTH - 2);
		int total = 0;

		while(count >= 0) {
			char value = creditCardNumber.charAt(count);
			String stringOfNumber = String.valueOf(value);
			int number = Integer.parseInt(stringOfNumber);

			number = (number * 2);

			if(number >= 10) {

				int firstNumber = (number % 10);   
				int secondNumber = (number / 10); 
				int sumOfDigits = (firstNumber + secondNumber);
				total = (total + sumOfDigits);

			} else {

				total = total + number;
			}
			
			
			count = count - 2;
		}
	
	return total; 
}
	public int sumNumbersInOddPositions(String creditCardNumber) {
		final int STRING_LENGTH = creditCardNumber.length();
		int count = 0;
		int totalOfNumbersInOddPositions = 0;

		while(count < STRING_LENGTH) {

			char value = creditCardNumber.charAt(count);
			String stringOfNumber = String.valueOf(value);
			int number = Integer.parseInt(stringOfNumber);
			if(count % 2 != 0) {
				totalOfNumbersInOddPositions = totalOfNumbersInOddPositions + number;
			}
			count = count + 1;
		}
	return totalOfNumbersInOddPositions;
}
	public String cardType(String creditCardNumber) {			////"4388576018402626"
		char firstType = creditCardNumber.charAt(0);
		char secondType = creditCardNumber.charAt(0);
		char thirdTypeFirstDigit = creditCardNumber.charAt(0);
		char thirdTypeSecondDigit = creditCardNumber.charAt(1);
		char fourthType = creditCardNumber.charAt(0);
		
		int firstCardDigit = Integer.parseInt(String.valueOf(firstType));
		int secondCardDigit = Integer.parseInt(String.valueOf(secondType));
		int thirdCardFirstNumber = Integer.parseInt(String.valueOf(thirdTypeFirstDigit));
		int thirdCardSecondNumber = Integer.parseInt(String.valueOf(thirdTypeSecondDigit));
		int fourthCardDigit = Integer.parseInt(String.valueOf(fourthType));

		if(firstCardDigit == 4) {
			return "VisaCard";

		} else if(secondCardDigit == 5) {
			return "MasterCard";

		} else if(thirdCardFirstNumber == 3 && thirdCardSecondNumber == 7) {
			return "American Express Cards";

		} else if(fourthCardDigit == 6) {
			return "Discover Cards";
		} else {
			return "Invalid Card";
		}
}
	public String[] card(String creditCardNumber) {
		
	ValidationFunction feedback = new ValidationFunction();
	
	String theTypeOfCard = feedback.cardType(creditCardNumber);

	int firstValue = feedback.doubleEverySecondNumber(creditCardNumber);
	int secondValue = feedback.sumNumbersInOddPositions(creditCardNumber);

	int sumOftheFirstAndSecondValues = (firstValue + secondValue);
	
	final int STRING_LENGTH = creditCardNumber.length();
	String creditCardDigitLength = String.valueOf(STRING_LENGTH);
	
	String message = "";
	if(sumOftheFirstAndSecondValues % 10 == 0) {
		message = "Credit Card Is Valid";
	} else {
		message = "Credit Card Is invalid";
	}
	
	String[] cardData = {theTypeOfCard, creditCardNumber, creditCardDigitLength, message};
	
	return cardData;
}
}